packages/design-system/src/organisms/formStepHeader/index.tsx

```diff
@@ -93,38 +108,61 @@ export const FormStepHeader: React.FC<FormStepHeaderProps> = ({
   steps,
   currentStep,
 }) => {
-  const listRef = useRef<FlatList>(null)
+  const shouldChangePosition = useBreakpointValue({
+    base: true,
+    md: false,
+  })
+  const [stepsWidth, setStepsWidth] = useState<Record<string, number>>({})
+
   const currentStepIndex = useMemo(
     () => steps.findIndex((step) => step.id === currentStep),
     [steps, currentStep],
   )
-  const onScrollToIndexFailed = useMemo(
-    () => makeScrollToIndexFailedHandler(listRef),
-    [listRef],
-  )
 
-  useEffect(() => {
-    listRef.current?.scrollToIndex({ index: currentStepIndex })
-  }, [currentStepIndex])
+  const updateWidth = useCallback((id: string, width: number) => {
+    setStepsWidth((prev) => ({ ...prev, [id]: width }))
+  }, [])
+
+  const position = useMemo(() => {
+    let position_ = 0
+    for (let i = 0; i < currentStepIndex; i++) {
+      position_ +=
+        stepsWidth[steps[i].id] +
+        SEPARATOR_WIDTH +
+        SEPARATOR_HORIZONTAL_MARGIN * 2
+    }
+    if (position_) {
+      position_ -= SEPARATOR_WIDTH / 2
+    }
+    return position_
+  }, [stepsWidth, currentStepIndex, steps])
```

Comment:
This function will be very hard to understand in the future.

```suggestion
  const stepXPosition = useMemo(() => {
    const initialPosition = 0
    const calculatedPosition = Array(currentStepIndex).reduce((position, _, index) => {
      return (
        position +
        stepsWidth[steps[i].id] + // Include width of the previous steps
        SEPARATOR_WIDTH + // Include width of the separator coming after a step
        SEPARATOR_HORIZONTAL_MARGIN * 2 // Include horizontal margin doubled (because it's on left & right)
      )
    } ,initialPosition)

    // Ignore separator width for the last step (it has no separator after)
    // Avoid negative values by using Math.max
    return Math.max(initialPosition, calculatedPosition - SEPARATOR_WIDTH / 2)
  }, [stepsWidth, currentStepIndex, steps])
```

Reply:
I could make a mistake in the code above. Could you test/suggest an alternative?